src/main/java/edu/ucsb/cs56/mapache_search/UserController.java

```diff
@@ -36,22 +43,43 @@ public String index(Model model) {
         return "user/index";
     }
 
+    // cron = 0 0 12 * * ? does task at midnight
+    // every ten seconds "*/10 * * * * *"
+    @Scheduled(cron = "0 0 12 * * ?")
+    public void resetCounter() {
+        Iterable<AppUser> users = userRepository.findAll();
+        for(AppUser u : users){
+            logger.info(Long.toString(u.getSearches()));
+            userRepository.findByUid(u.getUid()).get(0).setSearches(0l);
+        }
+            
+    } 
```

Comment:
Is this something we should do as a scheduled task? It probably won't scale well with many users, and will break if the app isn't online during midnight for whatever reason.

It would probably be a better idea to store the date of the user's last search and decide whether or not to reset the counter the next time the user performs a search. It would be far easier to test with this approach, as well.

Reply:
@btk5h  is absolutely right on this one.

This will rack up a huge bill on Heroku that I'm not willing to pay.  The free tier is based on an assumption that you don't use up the CPU 24x7, which this definitely would if/when we have even a modest number of users.

I like that fact that you researched how to do a cron job in the code, but this is not a good use of that.